lib/shopify_api/rest/request.ex

```diff
@@ -162,4 +162,62 @@ defmodule ShopifyAPI.REST.Request do
          {:ok, body} <- map_fetched,
          do: body
   end
+
+  @doc """
+  Take an existing URI and add additional params, appending and replacing as necessary
+  ## Examples
+      iex> add_params_to_url("http://example.com/wat", [])
+      "http://example.com/wat"
+      iex> add_params_to_url("http://example.com/wat", [q: 1])
+      "http://example.com/wat?q=1"
+      iex> add_params_to_url("http://example.com/wat", [q: 1, t: 2])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat", %{q: 1, t: 2})
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&t=2", [])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1", [t: 2])
+      "http://example.com/wat?q=1&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1", [q: 3, t: 2])
+      "http://example.com/wat?q=3&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&s=4", [q: 3, t: 2])
+      "http://example.com/wat?q=3&s=4&t=2"
+      iex> add_params_to_url("http://example.com/wat?q=1&s=4", %{q: 3, t: 2})
+      "http://example.com/wat?q=3&s=4&t=2"
+  """
+  @spec add_params_to_url(binary, list | map) :: binary
+  defp add_params_to_url(url, params) when is_map(params) do
```

Comment:
Now that this function is private, the compiler issues a warning about the docs for it being discarded. I think it is best to remove the docs, as the function should work as expected. Also, and somewhat related, we currently don't run doctests in our this repo, yet almost all of the files modified in this PR use the doctest syntax. It seems as though they were written more as typespecs than specific tests. I personally find them distracting and difficult to parse and would much prefer typespecs for annotating functions. Do other folks have any thoughts on this?

Reply:
I'm OK with removing the docs here. For the REST endpoints I think they can be helpful as more explicit usage examples, but currently they're a bit verbose and even worse: sometimes incorrect. I think my preference would be to leave them for now and reconsider when other changes have settled down a bit.